runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingTimersOverlay.java

```diff
@@ -0,0 +1,135 @@
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ProgressPieComponent;
+
+class WoodcuttingTimersOverlay extends Overlay
+{
+	// Range of Trees respawn time
+	static final int TREE_MAX_RESPAWN_TIME = 60;
+	private static final int TREE_MIN_RESPAWN_TIME = 30;
+	private static final float TREE_RANDOM_PERCENT_THRESHOLD = (float) TREE_MIN_RESPAWN_TIME / TREE_MAX_RESPAWN_TIME;
+	private static final Color DARK_GREEN = new Color(0, 100, 0);
+
+	private final Client client;
+	private final WoodcuttingPlugin plugin;
+
+	@Inject
+	private WoodcuttingTimersOverlay(Client client, WoodcuttingPlugin plugin)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.plugin = plugin;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		List<TreeRespawn> respawns = plugin.getRespawns();
+		if (respawns.isEmpty())
+		{
+			return null;
+		}
+
+		Instant now = Instant.now();
+		for (Iterator<TreeRespawn> it = respawns.iterator(); it.hasNext();)
+		{
+			Color pieFillColor = Color.YELLOW;
+			Color pieBorderColor = Color.ORANGE;
+			TreeRespawn treeRespawn = it.next();
+			float percent = (now.toEpochMilli() - treeRespawn.getStartTime().toEpochMilli()) / (float) treeRespawn.getRespawnTime();
+			WorldPoint worldPoint = treeRespawn.getWorldPoint();
+			LocalPoint loc = LocalPoint.fromWorld(client, worldPoint);
+
+			// This is used to get the X and Y for centre location to work
+			LocalPoint tempLoc = LocalPoint.fromWorld(client, worldPoint.getX(), worldPoint.getY());
+			LocalPoint centreLoc = new LocalPoint(
+				tempLoc.getX() + Perspective.LOCAL_TILE_SIZE * (client.getPlane() - 1) / 2,
+				tempLoc.getY() + Perspective.LOCAL_TILE_SIZE * (client.getPlane() - 1) / 2);
+			if (loc == null || percent > 1.0f)
+			{
+				it.remove();
+				continue;
+			}
+
+			Point centrePoint = Perspective.localToCanvas(client, centreLoc, client.getPlane());
+			Point point = Perspective.localToCanvas(client, loc, client.getPlane());
+			if (point == null || centrePoint == null)
+			{
+				it.remove();
+				continue;
+			}
+
+			Tree tree = treeRespawn.getTree();
+
+			// Recolour pie on regular trees within the range they spawn
+			if (tree == Tree.REGULAR_TREES && percent > TREE_RANDOM_PERCENT_THRESHOLD)
+			{
+				pieFillColor = Color.GREEN;
+				pieBorderColor = DARK_GREEN;
+			}
+
+			// The pie is already centred on these trees
+			if (tree == Tree.OAK_TREES || tree == Tree.YEW_TREES || tree == Tree.MAHOGANY_TREES || tree == Tree.TEAK_TREES || tree == Tree.REDWOODS)
+			{
+				ProgressPieComponent ppc = new ProgressPieComponent();
+				ppc.setBorderColor(pieBorderColor);
+				ppc.setFill(pieFillColor);
+				ppc.setPosition(point);
```

Comment:
All of the code in this block is duplicated below, with the single difference being `setPosition(point)` versus `setPosition(centrePoint)`. Please de-duplicate this.

Reply:
It is like this because some trees are on a different offset than others which makes the timer to display incorrectly like the magic tree stump. Although there could be a better way to do it than what I've done.
@Nightfirecat 
![centred vs uncentred](https://user-images.githubusercontent.com/19593476/65580224-2796bf00-df71-11e9-9408-108c4d897cd9.jpg)